patient-connect/200_ImplementingSSR.md

```diff
@@ -0,0 +1,158 @@
+name: Implementing Server-Side Rendering (SSR) for Provider and Location Profile Pages (WIP 🚧)
+---
+
+# Implementing Server-Side Rendering (SSR) for Provider and Location Profile Pages (WIP 🚧)
+
+## Introduction
+**The Server-Side Rendering (SSR)** feature in Care Search is an optional enhancement that allows provider and location information to be included in the initial HTML response from your server. This improves the chances that search engines will index your profile pages correctly, as key content is included at the time of page load.
+
+> **Note: While SSR can help with SEO, many other factors (e.g., page speed, mobile-friendliness) influence SEO performance.**
+
+## Why Use SSR?
+By default, Care Search uses JavaScript to fetch and render location and provider information after the page has already loaded. This may not be ideal for search engines, which prefer to see all key content when the page first loads. SSR allows your server to insert the provider or location data directly into the HTML before it’s sent to the user’s browser.
+
+## How SSR Works
+- **Client-Side Rendering**: Normally, Care Search renders profile or location information by calling APIs from the browser **after** the page has loaded, and inserting the data into the `<section id="loyal-search"></section>` tag.
+- **Server-Side Rendering**: With SSR, you can make a server-side request to the SSR API to fetch the HTML for provider or location data **before** responding to the client. This pre-fetched content is inserted into the `#loyal-search` section in the HTML response.
+
+## Client-Side Rendering Fallback
+It’s highly recommended to keep the `https://connect.loyalhealth.com/client/search.bundle.js` script tag on the page, even when using SSR. In case the SSR request fails, the client-side rendering will execute as usual, ensuring that provider and location data still appear on the page.
+
+## Implementing SSR
+Normally, your site’s page for a provider or location would include an empty placeholder like this:
+```html
+<section id="loyal-search"></section>
+```
+With SSR, when your server receives a request for a provider or location page (e.g., `/find-care/provider/:entityId` or `/find-care/location/:entityId`), instead of returning the page with an empty `<section>`, it should call our SSR API to pre-load content:
+```
+GET https://api.loyalhealth.com/search-ssr/:clientId/:marketId/:profileId?/:entityType/profile/:entityId
+```
+- `clientId`: Your unique client ID.
+- `marketId`: Your market ID.
+- `profileId`: (Optional) The profile ID, if you're using a non-default profile.
+- `entityType`: Either `provider` or `location`, depending on the type of data you are fetching.
+- `entityId`: The ID of the provider or location.
+  
+The API will return a chunk of HTML that can be inserted into the `#loyal-search` section before sending the page back to the user’s browser.
+
+## Examples
+Below are examples of how to do this in different server-side environments.
+
```

Comment:
The API will return a chunk of HTML containing the necessary provider/location page content.